File: frontend/src/types/timeline.ts
export interface TimelineItem {
  type: "journal" | "metric" | "result" | "goal";
  id: string;
  title: string;
  summary: string;
  date: string;
  tags: string[];
  metadata: Record<string, unknown>;
}

export interface TimelineResponse {
  items: TimelineItem[];
  total: number;
  page: number;
  per_page: number;
}

File: backend/app/schemas/timeline.py
"""Schemas for the unified timeline."""

from __future__ import annotations

from pydantic import BaseModel


class TimelineItem(BaseModel):
    type: str
    id: str
    title: str
    summary: str
    date: str
    tags: list[str]
    metadata: dict

    model_config = {"from_attributes": True}


class TimelineResponse(BaseModel):
    items: list[TimelineItem]
    total: int
    page: int
    per_page: int
